Backend/app/rider-platform/rider-app/Main/src-read-only/API/Action/UI/CustomerReferral.hs

```diff
@@ -56,6 +70,57 @@ type API =
            Kernel.Types.APISuccess.APISuccess
   )
 
+data MyCustomException = PersonError | PersonStatsError
+  deriving (Show, Generic)
+
+instance ToJSON MyCustomException
+
+instance Exception MyCustomException
+
+instance ToServantErr MyCustomException where
+  status PersonError = status404
+  status PersonStatsError = status410 -- FXME
+
+instance
+  SanitizedUrl (sub :: Type) =>
+  SanitizedUrl (Throws r :> sub)
+  where
+  getSanitizedUrl _ = getSanitizedUrl (Proxy :: Proxy sub)
+
+instance (HasOpenApi api) => HasOpenApi (Throws MyCustomException :> api) where
+  toOpenApi _ =
+    let apiOpenApi = toOpenApi (Proxy @api)
+        errorResponses =
+          [ ( "404",
+              "Person not found"
+            ),
+            ( "410",
+              "Person stats not found"
+            )
+          ]
+        updatedPaths = addErrorResponses (_openApiPaths apiOpenApi) errorResponses
+     in apiOpenApi {_openApiPaths = updatedPaths}
+
+addErrorResponses :: InsOrd.InsOrdHashMap FilePath PathItem -> [(Text, Text)] -> InsOrd.InsOrdHashMap FilePath PathItem
+addErrorResponses hashMapPaths resp =
+  InsOrd.mapWithKey (addResponses resp) hashMapPaths
+
+addResponses :: [(Text, Text)] -> FilePath -> PathItem -> PathItem
+addResponses resp _pathKey pathItem =
+  pathItem {_pathItemGet = updateOperation (_pathItemGet pathItem)}
+  where
+    updateOperation :: Maybe Operation -> Maybe Operation
+    updateOperation Nothing = Nothing
+    updateOperation (Just op) = Just $ op {_operationResponses = updatedResponses}
+      where
+        updatedResponses = foldr addResponse (_operationResponses op) resp
+
+addResponse :: (Text, Text) -> Responses -> Responses
+addResponse (code, _descript) resp =
+  let statusCode = Kernel.Prelude.read (T.unpack code) :: HttpStatusCode
```

Comment:
don't use partial `read` function, its better to have codes as `Int`, not `Text`